core/src/main/java/io/grpc/internal/AbstractManagedChannelImplBuilder.java

```diff
@@ -54,9 +66,35 @@
   private Executor executor;
   private final List<ClientInterceptor> interceptors = new ArrayList<ClientInterceptor>();
 
+  private final URI target;
+
+  @Nullable
+  private final SocketAddress directServerAddress;
+
   @Nullable
   private String userAgent;
 
+  @Nullable
+  private NameResolver nameResolver;
+
+  @Nullable
+  private LoadBalancer loadBalancer;
+
+  protected AbstractManagedChannelImplBuilder(URI target) {
+    this.target = Preconditions.checkNotNull(target);
+    this.directServerAddress = null;
+  }
+
+  protected AbstractManagedChannelImplBuilder(SocketAddress directServerAddress) {
```

Comment:
Maybe temporarily. But long-term it should probably be its own naming system: `inprocess:///nameOfServer`. I'll look deeper into things.